DigitalLearningSolutions.Web/Controllers/TrackingSystem/CentreConfiguration/RegistrationPromptsController.cs

```diff
@@ -71,32 +89,92 @@ public IActionResult EditRegistrationPrompt(EditRegistrationPromptViewModel mode
             {
                 SaveAction => EditRegistrationPromptPostSave(model),
                 AddPromptAction => RegistrationPromptAnswersPostAddPrompt(model),
+                BulkAction => EditRegistrationPromptBulk(model),
                 _ => RedirectToAction("Error", "LearningSolutions")
             };
         }
 
         [HttpGet]
-        [Route("Add/SelectPrompt")]
-        public IActionResult AddRegistrationPromptSelectPrompt()
+        [Route("{promptNumber:int}/Edit/Bulk")]
+        [ServiceFilter(typeof(RedirectEmptySessionData<EditRegistrationPromptData>))]
+        public IActionResult EditRegistrationPromptBulk(int promptNumber)
         {
-            var addRegistrationPromptData = TempData.Peek<AddRegistrationPromptData>();
+            var data = TempData.Peek<EditRegistrationPromptData>()!;
 
-            if (addRegistrationPromptData == null || !Request.Cookies.ContainsKey(CookieName))
+            var model = new BulkRegistrationPromptAnswersViewModel(
+                data.EditModel.OptionsString,
+                false,
+                promptNumber
+            );
+
+            return View("BulkRegistrationPromptAnswers", model);
+        }
+
+        [HttpPost]
+        [Route("Edit/Bulk")]
+        [ServiceFilter(typeof(RedirectEmptySessionDataXor<AddRegistrationPromptData, EditRegistrationPromptData>))]
+        public IActionResult EditRegistrationPromptBulkPost(BulkRegistrationPromptAnswersViewModel model)
+        {
+            ValidateBulkOptionsString(model.OptionsString);
+            if (!ModelState.IsValid)
             {
-                addRegistrationPromptData = new AddRegistrationPromptData();
-                var id = addRegistrationPromptData.Id;
-
-                Response.Cookies.Append(
-                    CookieName,
-                    id.ToString(),
-                    new CookieOptions
-                    {
-                        Expires = DateTimeOffset.UtcNow.AddDays(1)
-                    }
-                );
-                TempData.Set(addRegistrationPromptData);
+                return View("BulkRegistrationPromptAnswers", model);
             }
 
+            var editData = TempData.Peek<EditRegistrationPromptData>()!;
+            editData.EditModel!.OptionsString = model.OptionsString;
+            TempData.Set(editData);
+
+            return RedirectToAction("EditRegistrationPrompt", new { promptNumber = model.PromptNumber });
+        }
+
+        [HttpPost]
+        [Route("Add/Bulk")]
+        [ServiceFilter(typeof(RedirectEmptySessionDataXor<AddRegistrationPromptData, EditRegistrationPromptData>))]
+        public IActionResult AddRegistrationPromptBulkPost(BulkRegistrationPromptAnswersViewModel model)

```

Comment:
Super minor, but do you think it's worth grouping this with the Add/Bulk GET endpoint?